torch/distributed/distributed_c10d.py

```diff
@@ -73,7 +76,23 @@ def __new__(cls, name):
                              "Gloo or MPI backend for collective operations "
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
-            raise ValueError("Invalid backend: '{}'".format(name))
+            try:
+                if name not in sys.modules:
+                    # using "pytorch_" prefix to avoid module naming conflicts/confusion 
+                    third_lib = importlib.import_module("pytorch_{}".format(name))
```

Comment:
Extensions are typically named with a dash separator, e.g. `torch-ccl`.

Also, let's use the `torch` prefix instead of `pytorch`, given that PyTorch itself is published as `torch`.

Reply:
@pietern , can "torch" prefix make any confusion for users as it looks like module for old Torch?

Reply:
This is the name of the PyTorch package.

Also, it a Python package, so there's little ambiguity :)

Reply:
from search in pypi packages for "torch-" and "pytorch-" prefixes I see that both are used for PyTorch modules
if you suggest to use "torch-" then ok, it will make name shorter :)

Reply:
> Extensions are typically named with a dash separator, e.g. `torch-ccl`.

@pietern , the problem we met is cpp_extension.load() is unable to tolerate "-" character in "name" argument. A similar issue was raised at pytorch/extension-cpp#13. To workaround this, we have to propose "torch_xyz" as extension name rather than "torch-xyz".

when user installs cpp_extension by setup.py, he can specify different extension name and module name to workaound this.

ext_modules = [
    CppExtension(
        name="torch_xyz",
        ...
    )]

setup(
    name='torch-xyz',
    ...)
Through this way, "pip install torch-xyz" and "import torch_xyz" all work.